lua/entities/gmod_wire_dhdd.lua

```diff
@@ -14,11 +14,33 @@ function ENT:Initialize()
 	self.Outputs = WireLib.CreateOutputs( self, { "Memory [ARRAY]", "Size" } )
 	self.Inputs = WireLib.CreateInputs( self, { "Data [ARRAY]", "Clear", "AllowWrite" } )
 
-	self.Memory = {}
+	self:SetMemory()
 	self.ROM = false
 	self.AllowWrite = true
+end
+
+function ENT:SetMemory(Memory)
```

Comment:
The proxy table isn't specific to the current value of `self.Memory`, only the current value of `self`.